### PR TITLE
Include step resources being executed as part of the state

### PR DESCRIPTION
**Pull request by cquiroz**
We want to send the state of each subsystem when they are triggered individually.
This requires some changes to the engine and server to carry the state of each subsystem all the way to the client
Note that wee are  only sending the state for the current atom

### CODE REVIEW

File: modules/web/client/src/main/scala/observe/ui/components/services/ServerEventHandler.scala [hunk 1]
```diff
@@ -71,14 +83,14 @@ trait ServerEventHandler:
...
                   executionStatus.copy(
-                    configStatus =
-                      List[Resource | Instrument](Resource.TCS, Resource.Gcal, Instrument.GmosNorth)
-                        .foldLeft(executionStatus.configStatus)((cs, ss) =>
-                          cs.updatedWith(ss)(
-                            _.orElse(old.get(obsId).flatMap(_.configStatus.get(ss)))
-                              .orElse(ActionStatus.Pending.some)
-                          )
-                        )
+                    stepResources = Nil // TODO Fixme
```
Comment (cquiroz) [inline modules/web/client/src/main/scala/observe/ui/components/services/ServerEventHandler.scala:86]: This is an exercise for @rpiaggio 😄

**Review by jluhrs — COMMENTED** [review]
I don't really understand why this change is needed.

File: modules/web/client/src/main/scala/observe/ui/components/services/ServerEventHandler.scala (hunk 1 above)
Comment (rpiaggio) [reply modules/web/client/src/main/scala/observe/ui/components/services/ServerEventHandler.scala:86]: Lol, ok

File: modules/engine/src/main/scala/observe/engine/Engine.scala [hunk 2]
```diff
@@ -96,14 +95,10 @@ class Engine[F[_]: MonadThrow: Logger, S, U] private (
...
-               case ActionType.Configure(sys) => sys.some
-               case _                         => none
-             }
-    } yield (act.gen, sys)
-
-    x.map { case (p, s) =>
-      modifyS(c.sid)(u => u.startSingle(c.actCoords, Response.Configured(s))) *>
+    } yield act.gen
+
+    x.map { case p =>
```
Comment (jluhrs) [inline modules/engine/src/main/scala/observe/engine/Engine.scala:100]: case p ?

**Review by jluhrs — APPROVED** [review]
LGTM

File: modules/web/client/src/main/scala/observe/ui/components/Home.scala [hunk 3]
```diff
@@ -158,10 +160,11 @@ object Home:
                             else
                               // TODO We actually have to remember what step this belonged to before it was unselected.
                               (stepId.some,
-                               executionState.map(
-                                 ExecutionState.configStatus
-                                   .modify(_.map((k, _) => k -> ActionStatus.Pending))
-                               )
+                               executionState.map(identity)
+                               // TODO Fixme
```
Comment (cquiroz) [inline modules/web/client/src/main/scala/observe/ui/components/Home.scala:164]: Another exercise for @rpiaggio

**Pull request merged**